addons/misc/functions/fnc_createSuitcaseNuke.sqf

```diff
@@ -0,0 +1,239 @@
+            // Not strictly needed, but makes it possible to test in singleplayer
+            private _time = [{time}, {serverTime}] select isMultiplayer;
+
+            // If this code is 'locked' to the zeus's client, serverTime isn't needed
+            // But if other zeuses can modify the device, best to sync with serverTime
+            private _deltaTime = (call _time) - (_device getVariable [QGVAR(suitcaseNuke_time), (call _time)]);
+            _device setVariable [QGVAR(suitcaseNuke_time), (call _time), true];
+
+            _countdownRemaining = _countdownRemaining - _deltaTime;
+            if(_countdownRemaining < 0) then { _countdownRemaining = 0; };
+            _device setVariable [QGVAR(suitcaseNuke_countdown), _countdownRemaining, true];
+
+            // If the timer hits 0, trigger the device
+            if(_countdownRemaining <= 0) then {
+                _device setDamage 1;
+            };
+        };
+
+        // Add a flipflop to add some syncopation to the flashing message
+        private _flipFlop = (_device getVariable [QGVAR(suitcaseNuke_countdownFlipFlop), 0]);
+        private _deviceText = " ";
+        if(_flipFlop < 3) then {
+            if(_countdownRemaining >= (60*60)) then {
+                _deviceText = [_countdownRemaining, "HH:MM:SS"] call BIS_fnc_secondsToString;
+            } else {
+                _deviceText = [_countdownRemaining, "MM:SS"] call BIS_fnc_secondsToString;
+            };
+        };
+
+        // Set texture of each screen
+        { 
+            _device setObjectTextureGlobal[_x, 
+                format ['#(rgb,512,512,3)text(0,1,"LCD14",0.3,"#171717","#eb4034","\n%1")', _deviceText]
+            ]; 
+        } forEach [1,2,3];
+        
+        // If the device has been disarmed (or triggered), flash the countdown on and off
+        if(!alive _device || !_isArmed) then {
+            _device setVariable [QGVAR(suitcaseNuke_countdownFlipFlop), (_flipFlop+1)%4, true];
+        } else {
+            // In case device is re-armed while flipFlop=3
+            _device setVariable [QGVAR(suitcaseNuke_countdownFlipFlop), 0, true];
+        };
+        
+
+        // If the device is deleted, remove this EH
+        if(isNull _device) then {
+            _handle call CBA_fnc_removePerFrameHandler;
+        };
+
+    }, 0.3, [_device]] call CBA_fnc_addPerFrameHandler;
+
+
+    // Activation Code
+    _device setVariable [QGVAR(suitcaseNuke_activate), _effect, true];
+    _device addMPEventHandler ["MPKilled", {
+        params ["_device"];
+        if(isServer) then {
+            switch(_device getVariable QGVAR(suitcaseNuke_activate)) do {
+                case "emp": {
+                    if (isClass (configFile >> "CfgPatches" >> "crowsEW_main")) then {
+                        [getPos _device, _device, 500, false, 1, 1] call EMFUNC(crowsEW,emp,fireEMP);
+                    } else {
+                        diag_log localize "STR_CROWSZA_Misc_suitcaseNuke_error_ewarMismatch";
```

Comment:
don't localize diag_log messages. Those are mainly for us. If localised and someone share an rpt and its in chinese we won't know what to search/look for ;-)

Only need to localize player-facing messages/HUD elements